setup-sccache-dist/action.yml

```diff
@@ -0,0 +1,149 @@
+            SCHEDULER_URL="https://arm64.linux.sccache.rapids.nvidia.com"
+            ;;
+          *)
+            echo "Unsupported CPU architecture '$ARCH'" >&2
+            exit 1
+            ;;
+        esac
+
+        # Install sccache to /usr/bin
+        while ! tar -C /usr/bin -z --wildcards --strip-components=1 -x '*/sccache' -f <(
+          if [ "${VERSION:-latest}" = latest ]; then
+            # Install latest version
+            wget --no-hsts -q -O- https://api.github.com/repos/${REPO}/releases/latest \
+          | jq -r ".assets[] | select(.name | test(\"^sccache-v.*?-${ARCH}-unknown-linux-musl.tar.gz\$\")) | .browser_download_url" \
+          | wget --no-hsts -q -O- -i-
+          else
+            # Install pinned version
+            wget --no-hsts -q -O- "https://github.com/${REPO}/releases/download/v${VERSION}/sccache-v${VERSION}-${ARCH}-unknown-linux-musl.tar.gz"
+          fi
+        ); do
+          sleep 10;
+        done
+
+        # Verify installation
+        sccache --version
+        mkdir -p "$(dirname "$LOG_FILE")"
+        stat "$(dirname "$LOG_FILE")"
+        ulimit -n "$(ulimit -Hn)"
+        echo "nofile ulimit: $(ulimit -n):$(ulimit -Hn)"
+
+        # Runner configuration
+        export PARALLEL_LEVEL="$(ulimit -Hn)"
+        export GHA_CACHE_SLUG="${{github.run_id}}-${{github.run_attempt}}-$RANDOM"
+        export NVCC_APPEND_FLAGS="${NVCC_APPEND_FLAGS:+$NVCC_APPEND_FLAGS }-t=100"
+
+        # sccache configuration
+        source rapids-configure-sccache
+        export SCCACHE_IDLE_TIMEOUT="0"
+        export SCCACHE_ERROR_LOG="$LOG_FILE"
+        export SCCACHE_SERVER_LOG="$LOG_LEVEL"
+
+        # sccache-dist configuration
+        export SCCACHE_DIST_AUTH_TOKEN="$AUTH"
+        export SCCACHE_DIST_AUTH_TYPE="token"
+        export SCCACHE_DIST_FALLBACK_TO_LOCAL_COMPILE="$FALLBACK_TO_LOCAL_COMPILE"
+        export SCCACHE_DIST_MAX_RETRIES="$MAX_RETRIES"
+        export SCCACHE_DIST_REQUEST_TIMEOUT="$REQUEST_TIMEOUT"
+        export SCCACHE_DIST_SCHEDULER_URL="$SCHEDULER_URL"
+
+        cat <<EOF | tee -a "$GITHUB_ENV"
+        PARALLEL_LEVEL=$PARALLEL_LEVEL
+        GHA_CACHE_SLUG=$GHA_CACHE_SLUG
+        NVCC_APPEND_FLAGS=$NVCC_APPEND_FLAGS
+        SCCACHE_BUCKET=$SCCACHE_BUCKET
+        SCCACHE_REGION=$SCCACHE_REGION
+        SCCACHE_IDLE_TIMEOUT=$SCCACHE_IDLE_TIMEOUT
+        SCCACHE_ERROR_LOG=$SCCACHE_ERROR_LOG
+        SCCACHE_SERVER_LOG=$SCCACHE_SERVER_LOG
+        SCCACHE_DIST_AUTH_TOKEN=$SCCACHE_DIST_AUTH_TOKEN
+        SCCACHE_DIST_AUTH_TYPE=$SCCACHE_DIST_AUTH_TYPE
+        SCCACHE_DIST_FALLBACK_TO_LOCAL_COMPILE=$SCCACHE_DIST_FALLBACK_TO_LOCAL_COMPILE
+        SCCACHE_DIST_MAX_RETRIES=$SCCACHE_DIST_MAX_RETRIES
+        SCCACHE_DIST_REQUEST_TIMEOUT=$SCCACHE_DIST_REQUEST_TIMEOUT
+        SCCACHE_DIST_SCHEDULER_URL=$SCCACHE_DIST_SCHEDULER_URL
+        EOF
+
+        # Start the sccache client daemon
+        sccache --start-server
+
+    - id: sccache-preprocessor-cache
```

Comment:
This preprocessor cache is a good idea for a GitHub Action because it's exclusively needed in CI nodes. However, I would like to see the other logic (install `sccache` and set up environment variables) in [rapids-configure-sccache](https://github.com/rapidsai/gha-tools/blob/main/tools/rapids-configure-sccache), which is already called at the beginning of every CI build job. That way we can benefit from `sccache-dist` when performing local reproductions of the CI build workflow, too.

Reply:
Tested in https://github.com/rapidsai/rmm/pull/2101 (GHA job [here](https://github.com/rapidsai/rmm/actions/runs/19046487727?pr=2101))